3.processing_features/scripts/2.annotate_sc.py

```diff
@@ -62,11 +62,43 @@
 pprint.pprint(plate_info_dictionary, indent=4)
 
 
+# ## Annotate bulk profiles
+# 
+# **Note:** The path to the annotated bulk file to be used for normalization is adding during this step.
+
+# In[4]:
+
+
+for plate, info in plate_info_dictionary.items():
+    # single_cell_df is the dataframe loaded in from the converted parquet file
+    single_cell_df = pd.read_parquet(info["bulk_path"])
+    platemap_df = pd.read_csv(info["platemap_path"])
+    output_file = str(pathlib.Path(f"{output_dir}/{plate}_bulk_annotated.parquet"))
+    # save path to annotated file to dictionary for downstream use
+    plate_info_dictionary[plate]["bulk_annotated_path"] = output_file
+    print(f"Adding annotations to merged single cells for {plate}!")
+
+    # add metadata from platemap file to extracted single cell features
+    annotated_df = annotate(
+        profiles=single_cell_df,
+        platemap=platemap_df,
+        join_on=["Metadata_well_position", "Image_Metadata_Well"],
+    )
+
+    # save annotated df as parquet file
+    output(
+        df=annotated_df,
+        output_filename=output_file,
+        output_type="parquet",
+    )
+    print(f"Annotations have been added to {plate} bulk profiles and saved!")
+
+
 # ## Annotate merged single cells
```

Comment:
Same here, single cells or bulk? or aggregated single cells?

Reply:
Same as above, all of these processes are run on both bulk and sc so I have tried to make that less confusing.